### PR TITLE
Tracking headers

### PR DESCRIPTION
**Pull request by ewoutp**
This PS initializes the activity and command line headers, used for activity tracking.

Dependencies:

- [x] https://github.com/giantswarm/client-go/pull/95

Epic:

- https://github.com/giantswarm/giantswarm/issues/421

### CODE REVIEW

**Comment by ewoutp** [comment]
@marians I've now chosen to derive the activity from the sub command.
This results in activities like `swarm-cat` `swarm-ls` etc. If need we can override that, but this way we don't have to think about it for new commands.
It results log messages like this:

```
{"activity":"swarm-cat","cmdline":"swarm --api-endpoint=http://192.168.33.101:8080/v1/ --config-dir=/Users/ewout/.swarm-lc cat","request-id":"5wxymu996nwi6cxv, f5kvcvl7u5fk7hyb"}
```

**Comment by marians** [comment]
@ewoutp That's fine, however I'd like us to think of useful names.

Example: `swarm env` can be called to select an environment or to create a new one. I am currently working on the list for all existing activities.

**Comment by marians** [comment]
Sorry, it should have been: `swarm env` can be called to list environments or to select/create one.

**Comment by ewoutp** [comment]
I've added all tracking activities.

@marians @zeisss RFR

File: list.go [hunk 1]
```diff
@@ -84,8 +84,7 @@ func listServices() {
 				// For event environment, create a new client and get all services
 				for _, env := range envs.Environments {
 					// Create new client for this environment
-					envClient, err := createClient(globalFlags.ApiEndpoint, org+"/"+env.Name)
-					envClient.SetRequestID(requestId)
```
Comment (marians) [inline list.go]: Any specific reason to remove this `envClient.SetRequestID(requestId)` here?

Comment (marians) [inline list.go]: Ah, found it further below :)

**Comment by marians** [comment]
LGTM

**Comment by marians** [comment]
As discussed offline, we should add an opt-out mechanism via the environment variable `GIANTSWARM_DISABLE_CMDLINE_TRACKING`. If this is set to anything other than an empty string, the `X-Giant-Swarm-CmdLine` must not be set.

**Comment by ewoutp** [comment]
Added optout in case environment variable `GIANTSWARM_DISABLE_CMDLINE_TRACKING` is non-empty.

File: main.go [hunk 2]
```diff
@@ -265,3 +283,9 @@ func Verbosef(message string, s ...interface{}) {
 		Printf(message, s...)
 	}
 }
+
+// setnxActivity initializes the activity of the current client, unless it already
+// has an activity.
+func setnxActivity(activity string) {
```
Comment (zeisss) [inline main.go]: `setnxActivity` ? NX?

Comment (ewoutp) [inline main.go]: > unless it already has an activity.

Comment (zeisss) [inline main.go]: where is this logic? if its in the client package, shouldn't this method also have the `nx` part?

Comment (ewoutp) [inline main.go]: Big oops, thanks for catching

Comment (ewoutp) [inline main.go]: Please look again

**Comment by zeisss** [comment]
LGTM

**Pull request merged**